### ISSUE TITLE
Vulnerability Roundup 10

### ISSUE THREAD

**Issue by grahamc** [NixOS/nixpkgs#20647]
This is a special roundup!

1. We have now been doing this for two full months (since September 22)
2. run 10 roundups
3. have examined 997 vulnerabilities (!)

This is also special because it is the first time the tooling I've written can be open source, AND LWN has given me written permission to be using their database for this purpose. This tooling now lives at https://github.com/NixOS/security.

---

Here are all the vulnerabilities from https://lwn.net/Vulnerabilities
since our [last roundup](https://github.com/NixOS/nixpkgs/issues/20462).

cc: @FRidh, @fpletz, @NeQuissimus, and @phanimahesh.

_Note:_ The list of people CC'd on this issue participated in the last
roundup. If you participate on this roundup, I'll cc you on the next
one. If you don't participate in the next one, you won't be CC'd on
the one after that. If you would like to be CC'd on the next roundup,
add a comment to the most recent vulnerability roundup. If you
would like to be CC'd on _all_ roundups, leave a comment and tell
@grahamc so.

Permanent CC's: @joepie91, @NixOS/security-notifications
(if you no longer want to be CC'd, ask to be removed from this list)

## Notes on the list
1. The reports have been roughly grouped by the package name. This
   isn't perfect, but is intended to help identify if a whole group
   of reports is resolved already.
2. Some issues will be duplicated, because it affects multiple packages.
   For example, there are sometimes problems that impact thunderbird,
   and firefox. LWN might report in one vulnerability "thunderbird
   firefox". These names have been split to make sure both packages get
   addressed.
3. By each issue is a link to code search for the package name, and
   a Github search by filename. These are to help, but may not return
   results when we do in fact package the software. If a search
   doesn't turn up, please try altering the search criteria or
   looking in nixpkgs manually before asserting we don't have it.
4. This issue is created by https://github.com/NixOS/security

# Instructions:

1. Triage a report: If we don't have the software or our version isn't
   vulnerable, tick the box or add a comment with the report number,
   stating it isn't vulnerable.
2. Fix the issue: If we do have the software and it is vulnerable,
   either leave a comment on this issue saying so, even open a pull
   request with the fix. If you open a PR, make sure to tag this
   issue so we can coordinate.
3. When an entire section is completed, move the section to the
   "Triaged and Resolved Issues" `details` block below.

## Upon Completion ...

 - [x] Run the issue through `reformat` one last time
 - [x] Review commits since last roundup for backport candidates
 - [x] Update https://github.com/NixOS/nixpkgs/issues/13515 with a
 summary.
 - [x] Update the database at https://github.com/NixOS/security

Without further ado...

### Assorted (23 issues)
 - [x] [`#706586`](https://lwn.net/Vulnerabilities/706586/) ([search](http://search.nix.gsc.io/?q=gst-plugins-bad&i=fosho&repos=nixos-nixpkgs), [files](https://github.com/NixOS/nixpkgs/search?utf8=%E2%9C%93&q=gst-plugins-bad+in%3Apath&type=Code))  gst-plugins-bad0.10: code execution
 - [x] [`#706842`](https://lwn.net/Vulnerabilities/706842/) ([search](http://search.nix.gsc.io/?q=gst-plugins-bad1.0&i=fosho&repos=nixos-nixpkgs), [files](https://github.com/NixOS/nixpkgs/search?utf8=%E2%9C%93&q=gst-plugins-bad1.0+in%3Apath&type=Code))  gst-plugins-bad: code execution
 - [x] [`#706851`](https://lwn.net/Vulnerabilities/706851/) ([search](http://search.nix.gsc.io/?q=GraphicsMagick&i=fosho&repos=nixos-nixpkgs), [files](https://github.com/NixOS/nixpkgs/search?utf8=%E2%9C%93&q=GraphicsMagick+in%3Apath&type=Code))  graphicsmagick: denial of service
 - [x] [`#706582`](https://lwn.net/Vulnerabilities/706582/) ([search](http://search.nix.gsc.io/?q=akonadi&i=fosho&repos=nixos-nixpkgs), [files](https://github.com/NixOS/nixpkgs/search?utf8=%E2%9C%93&q=akonadi+in%3Apath&type=Code))  akonadi: denial of service
 - [x] [`#706577`](https://lwn.net/Vulnerabilities/706577/) ([search](http://search.nix.gsc.io/?q=atomic-openshift&i=fosho&repos=nixos-nixpkgs), [files](https://github.com/NixOS/nixpkgs/search?utf8=%E2%9C%93&q=atomic-openshift+in%3Apath&type=Code))  atomic-openshift: redirect network traffic
 - [x] [`#706846`](https://lwn.net/Vulnerabilities/706846/) ([search](http://search.nix.gsc.io/?q=libtiff&i=fosho&repos=nixos-nixpkgs), [files](https://github.com/NixOS/nixpkgs/search?utf8=%E2%9C%93&q=libtiff+in%3Apath&type=Code))  libtiff: multiple vulnerabilities
 - [x] [`#707046`](https://lwn.net/Vulnerabilities/707046/) ([search](http://search.nix.gsc.io/?q=qemu&i=fosho&repos=nixos-nixpkgs), [files](https://github.com/NixOS/nixpkgs/search?utf8=%E2%9C%93&q=qemu+in%3Apath&type=Code))  qemu: denial of service
 - [x] [`#707037`](https://lwn.net/Vulnerabilities/707037/) ([search](http://search.nix.gsc.io/?q=ipsilon&i=fosho&repos=nixos-nixpkgs), [files](https://github.com/NixOS/nixpkgs/search?utf8=%E2%9C%93&q=ipsilon+in%3Apath&type=Code))  ipsilon: information leak/denial of service
 - [x] [`#663078`](https://lwn.net/Vulnerabilities/663078/) ([search](http://search.nix.gsc.io/?q=krb5&i=fosho&repos=nixos-nixpkgs), [files](https://github.com/NixOS/nixpkgs/search?utf8=%E2%9C%93&q=krb5+in%3Apath&type=Code))  krb5: multiple vulnerabilities
 - [x] [`#638448`](https://lwn.net/Vulnerabilities/638448/) ([search](http://search.nix.gsc.io/?q=mongodb&i=fosho&repos=nixos-nixpkgs), [files](https://github.com/NixOS/nixpkgs/search?utf8=%E2%9C%93&q=mongodb+in%3Apath&type=Code))  mongodb: denial of service
 - [x] [`#707043`](https://lwn.net/Vulnerabilities/707043/) ([search](http://search.nix.gsc.io/?q=sniffit&i=fosho&repos=nixos-nixpkgs), [files](https://github.com/NixOS/nixpkgs/search?utf8=%E2%9C%93&q=sniffit+in%3Apath&type=Code))  sniffit: privilege escalation
 - [x] [`#707040`](https://lwn.net/Vulnerabilities/707040/) ([search](http://search.nix.gsc.io/?q=w3m&i=fosho&repos=nixos-nixpkgs), [files](https://github.com/NixOS/nixpkgs/search?utf8=%E2%9C%93&q=w3m+in%3Apath&type=Code))  w3m: multiple vulnerabilities
 - [x] [`#706844`](https://lwn.net/Vulnerabilities/706844/) ([search](http://search.nix.gsc.io/?q=gnuchess&i=fosho&repos=nixos-nixpkgs), [files](https://github.com/NixOS/nixpkgs/search?utf8=%E2%9C%93&q=gnuchess+in%3Apath&type=Code))  gnuchess: code execution
 - [x] [`#706588`](https://lwn.net/Vulnerabilities/706588/) ([search](http://search.nix.gsc.io/?q=libarchive&i=fosho&repos=nixos-nixpkgs), [files](https://github.com/NixOS/nixpkgs/search?utf8=%E2%9C%93&q=libarchive+in%3Apath&type=Code))  libarchive: unspecified
 - [x] [`#635283`](https://lwn.net/Vulnerabilities/635283/) ([search](http://search.nix.gsc.io/?q=libuv&i=fosho&repos=nixos-nixpkgs), [files](https://github.com/NixOS/nixpkgs/search?utf8=%E2%9C%93&q=libuv+in%3Apath&type=Code))  libuv: privilege escalation
 - [x] [`#706852`](https://lwn.net/Vulnerabilities/706852/) ([search](http://search.nix.gsc.io/?q=otrs&i=fosho&repos=nixos-nixpkgs), [files](https://github.com/NixOS/nixpkgs/search?utf8=%E2%9C%93&q=otrs+in%3Apath&type=Code))  otrs: code execution
 - [x] [`#707038`](https://lwn.net/Vulnerabilities/707038/) ([search](http://search.nix.gsc.io/?q=drupal&i=fosho&repos=nixos-nixpkgs), [files](https://github.com/NixOS/nixpkgs/search?utf8=%E2%9C%93&q=drupal+in%3Apath&type=Code))  drupal: multiple vulnerabilities
 - [x] [`#706734`](https://lwn.net/Vulnerabilities/706734/) ([search](http://search.nix.gsc.io/?q=firefox-esr&i=fosho&repos=nixos-nixpkgs), [files](https://github.com/NixOS/nixpkgs/search?utf8=%E2%9C%93&q=firefox-esr+in%3Apath&type=Code))  firefox: timing side channel
 - [x] [`#707045`](https://lwn.net/Vulnerabilities/707045/) ([search](http://search.nix.gsc.io/?q=moodle&i=fosho&repos=nixos-nixpkgs), [files](https://github.com/NixOS/nixpkgs/search?utf8=%E2%9C%93&q=moodle+in%3Apath&type=Code))  moodle: multiple vulnerabilities
 - [x] [`#706581`](https://lwn.net/Vulnerabilities/706581/) ([search](http://search.nix.gsc.io/?q=nss&i=fosho&repos=nixos-nixpkgs), [files](https://github.com/NixOS/nixpkgs/search?utf8=%E2%9C%93&q=nss+in%3Apath&type=Code))  nss, nss-util: two vulnerabilities
 - [x] [`#706581`](https://lwn.net/Vulnerabilities/706581/) ([search](http://search.nix.gsc.io/?q=nss-util&i=fosho&repos=nixos-nixpkgs), [files](https://github.com/NixOS/nixpkgs/search?utf8=%E2%9C%93&q=nss-util+in%3Apath&type=Code))  nss, nss-util: two vulnerabilities
 - [x] [`#707039`](https://lwn.net/Vulnerabilities/707039/) ([search](http://search.nix.gsc.io/?q=php&i=fosho&repos=nixos-nixpkgs), [files](https://github.com/NixOS/nixpkgs/search?utf8=%E2%9C%93&q=php+in%3Apath&type=Code))  php: code execution
 - [x] [`#706848`](https://lwn.net/Vulnerabilities/706848/) ([search](http://search.nix.gsc.io/?q=wireshark&i=fosho&repos=nixos-nixpkgs), [files](https://github.com/NixOS/nixpkgs/search?utf8=%E2%9C%93&q=wireshark+in%3Apath&type=Code))  wireshark: multiple vulnerabilities

### drupal7 (2 issues)
 - [x] [`#707041`](https://lwn.net/Vulnerabilities/707041/) ([search](http://search.nix.gsc.io/?q=drupal7&i=fosho&repos=nixos-nixpkgs), [files](https://github.com/NixOS/nixpkgs/search?utf8=%E2%9C%93&q=drupal7+in%3Apath&type=Code))  drupal7: URL injection
 - [x] [`#706841`](https://lwn.net/Vulnerabilities/706841/) ([search](http://search.nix.gsc.io/?q=drupal7&i=fosho&repos=nixos-nixpkgs), [files](https://github.com/NixOS/nixpkgs/search?utf8=%E2%9C%93&q=drupal7+in%3Apath&type=Code))  drupal: multiple vulnerabilities

### firefox (3 issues)
 - [x] [`#707047`](https://lwn.net/Vulnerabilities/707047/) ([search](http://search.nix.gsc.io/?q=firefox&i=fosho&repos=nixos-nixpkgs), [files](https://github.com/NixOS/nixpkgs/search?utf8=%E2%9C%93&q=firefox+in%3Apath&type=Code))  mozilla: code execution
 - [x] [`#706580`](https://lwn.net/Vulnerabilities/706580/) ([search](http://search.nix.gsc.io/?q=firefox&i=fosho&repos=nixos-nixpkgs), [files](https://github.com/NixOS/nixpkgs/search?utf8=%E2%9C%93&q=firefox+in%3Apath&type=Code))  firefox: multiple vulnerabilities
 - [x] [`#706731`](https://lwn.net/Vulnerabilities/706731/) ([search](http://search.nix.gsc.io/?q=firefox&i=fosho&repos=nixos-nixpkgs), [files](https://github.com/NixOS/nixpkgs/search?utf8=%E2%9C%93&q=firefox+in%3Apath&type=Code))  firefox: multiple vulnerabilities

### kernel (2 issues)
 - [x] [`#706735`](https://lwn.net/Vulnerabilities/706735/) ([search](http://search.nix.gsc.io/?q=kernel&i=fosho&repos=nixos-nixpkgs), [files](https://github.com/NixOS/nixpkgs/search?utf8=%E2%9C%93&q=kernel+in%3Apath&type=Code))  kernel: denial of service
 - [x] [`#707044`](https://lwn.net/Vulnerabilities/707044/) ([search](http://search.nix.gsc.io/?q=kernel&i=fosho&repos=nixos-nixpkgs), [files](https://github.com/NixOS/nixpkgs/search?utf8=%E2%9C%93&q=kernel+in%3Apath&type=Code))  kernel: denial of service

**Comment by fpletz** [comment]
PHP and wireshark were updated recently to fix the mentioned issues.

Patch for [#706735](https://lwn.net/Vulnerabilities/706735/) is in Linux kernel 4.8.10 (which we already have) but looks like the other issue isn't patched already.

**Comment by grahamc** [comment]
Oh hrm I already marked the kernel as done. Unchecked 706735.

**Comment by fpletz** [comment]
I'm building qemu right now with the patch for CVE-2016-7907. :wink:

**Comment by phanimahesh** [comment]
A bit occupied atm, will get to these as soon as I can, can take a while. Since this is my first, it may take some time for me to familiarize myself with the process.

Also @grahamc, `If you would like to be CC'd on the next roundup` is repeated twice in the message.

**Comment by grahamc** [comment]
According to https://security-tracker.debian.org/tracker/CVE-2016-8862, the graphicsmagick issue here is due to a bad patch for another CVE.

**Comment by grahamc** [comment]
Libarchive needs an update, have it staged here: https://github.com/NixOS/nixpkgs/pull/20668

**Comment by grahamc** [comment]
Gnuchess in the rollup.

**Comment by grahamc** [comment]
w3m in the rollup.

**Comment by grahamc** [comment]
krb5 was an old announcement we already patched against.

**Comment by grahamc** [comment]
Mongodb needs updating to .11, about to push a patch to my rollup.

**Comment by grahamc** [comment]
I was mistaken, our version is _not_ impacted by the mongodb issue reported.

**Comment by grahamc** [comment]
The Akonadi issue appears to be configuration related on debian.

**Comment by grahamc** [comment]
@globin, @dezgeg, @ttuegel -- it looks like there are some security updates in gstreamer 1.10.1. 16.09 is running 1.8.x. Is this a safe update? How should we handle this?

**Comment by grahamc** [comment]
@fpletz I just pushed your qemu patch to my roundup branch, and will backport it as well.

**Comment by grahamc** [comment]
Updated libtiff to 4.0.7 which fixed all our patches + some more CVEs.

**Comment by NeQuissimus** [comment]
Good work guys. Sorry, I could not be of any help this time around...

**Comment by grahamc** [comment]
That is OK, @NeQuissimus :) can you check in to the kernel thing?

**Comment by NeQuissimus** [comment]
That CVE was fixed in 4.8.7 (https://cdn.kernel.org/pub/linux/kernel/v4.x/ChangeLog-4.8.7) and 4.4.31 (https://cdn.kernel.org/pub/linux/kernel/v4.x/ChangeLog-4.4.31), respectively. We are good on both 4.8.x and 4.4.x kernels.

**Comment by grahamc** [comment]
graphicsmagick seems to be fine.

**Comment by fpletz** [comment]
@grahamc I did the gstreamer update to 1.10.1 due to the security issues in 7a6185d9a1eb18fa09892fdef01a5193968cf704 but didn't backport it yet because I wasn't sure if it breaks anything. I did some testing but didn't have the time to build every dependency.

The [changelog](https://gstreamer.freedesktop.org/releases/1.10/) doesn't mention any breaking or incompatible changes, though, just additions to the API. The note about packaging is not relevant for us because we don't have architecture-independent expressions. So this might actually be quite safe. :+1:
